sys/include/uart_stdio.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_uart_stdio UART stdio
+ * @ingroup     sys
+ *
+ * @brief       stdio init/read/write functions for UARTs
+ *
+ * @{
+ * @file
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+#ifndef UART_STDIO_H
+#define UART_STDIO_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief initialize the module
+ */
+void uart_stdio_init(void);
+
+/**
+ * @brief read len bytes from stdio uart into buffer
+ */
+int uart_stdio_read(char* buffer, int len);
```

Comment:
But by letting doxygen alert on missing parameter documentation we can make
sure that the api documents are always up to date with regards to API
changes.
On Jul 7, 2015 4:36 PM, "Kaspar Schleiser" <notifications@github.com> wrote:

> In sys/include/uart_stdio.h
> <https://github.com/RIOT-OS/RIOT/pull/3161#discussion_r34046394>:
>
> > +#ifndef UART_STDIO_H
> > +#define UART_STDIO_H
> > +
> > +#ifdef __cplusplus
> > +extern "C" {
> > +#endif
> > +
> > +/**
> > + * @brief initialize the module
> > + */
> > +void uart_stdio_init(void);
> > +
> > +/**
> > + * @brief read len bytes from stdio uart into buffer
> > + */
> > +int uart_stdio_read(char* buffer, int len);
>
>  I think Doxygen will still warn about undocumented parameters.
> Again we're writing documentation for Doxygen, not our highly valued users.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RIOT-OS/RIOT/pull/3161/files#r34046394>.
>